### PR TITLE
[Naitei] [Ecommerce] [Admin] Update category

### PR DESCRIPTION
**Pull request by SuKay712**
## Related Tickets
https://edu-redmine.sun-asterisk.vn/issues/78412

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2024-07-28 11-50-00

Screenshot from 2024-07-28 11-57-16

Screenshot from 2024-07-28 11-53-12

### CODE REVIEW

**Comment by SuKay712** [comment]
ready

File: app/views/admin/categories/_form.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<%= form_with(model: @category, url: admin_categories_path) do |form| %>
-  <%= render "shared/error_messages", object: form.object %>
-  <%= form.label :name, t("category.name") %>
+<%= form_with(model: @category, url: url, method: method) do |form| %>
+  <%= render "shared/error_messages", object: form.object%>
```
Comment (ThaiKhacDuoc) [inline app/views/admin/categories/_form.html.erb]: Thiếu space huynh ơi
```suggestion
  <%= render "shared/error_messages", object: form.object %>
```

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -19,8 +20,28 @@ def create
...
   private
   def category_params
     params.require(:category).permit(:name, :parent_category_id)
   end
+
+  def load_category
+    @category = Category.find_by id: params[:id]
+    return if @category
+
+    flash[:danger] = t("flash.create_unsuccessfully")
```
Comment (ThaiKhacDuoc) [inline app/controllers/admin/categories_controller.rb]: Bạn bỏ () cho đồng bộ file này nha
`t("flash.create_unsuccessfully")`

**Review by ThaiKhacDuoc — COMMENTED** [review]
cmt

File: app/controllers/admin/categories_controller.rb (hunk 2 above)
Comment (SuKay712) [reply app/controllers/admin/categories_controller.rb]: oke

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**